src/sql/QueryEditor/FillValueSelect.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { DataQuery, SelectableValue } from '@grafana/data';
+import { InlineField, Input, Select } from '@grafana/ui';
+
+export enum FillValueOptions {
+  Previous,
+  Null,
+  Value,
+}
+
+export const SelectableFillValueOptions: Array<SelectableValue<FillValueOptions>> = [
+  {
+    label: 'Previous Value',
+    value: FillValueOptions.Previous,
+  },
+  {
+    label: 'NULL',
+    value: FillValueOptions.Null,
+  },
+  {
+    label: 'Value',
+    value: FillValueOptions.Value,
+  },
+];
+
+export type FillValueSelectProps<TQuery extends DataQuery> = {
+  query: TQuery;
+  onChange: (value: TQuery) => void;
+  onRunQuery: () => void;
+};
+
+export function FillValueSelect<TQuery extends DataQuery & Record<string, any>>(props: FillValueSelectProps<TQuery>) {
+  return (
+    <>
+      <InlineField label="Fill value" tooltip="value to fill missing points">
+        <Select
+          aria-label="Fill value"
+          options={SelectableFillValueOptions}
+          value={props.query.fillMode?.mode ?? FillValueOptions.Previous}
+          onChange={({ value }) => {
+            props.onChange({
+              ...props.query,
+              fillMode: { mode: value, value: props.query.fillMode?.value },
```

Comment:
I am probably misunderstanding but shouldn't the value be set to different things depending on the mode rather than whatever was last saved in the query?

Reply:
The only mode that uses the value is `FillValueOptions.Value` (the third one), so this is to avoid losing the value set there when changing modes. Added a comment to explain that.

Reply:
ah I see. I guess you could decouple them then? But I suppose a comment is probably ok for now!

Reply:
what do you mean by decoupling them? (I guess not setting the value?)

Reply:
I meant like you have both pieces of state in the same object, so when you set one property you need to carry along the other property, even though when one changes you want the other to stay the same so there's no strong need to keep them together. That way you only need to set the value when it changes. 

This is just nitpicking code style like stuff on my part though. How you have it is great! I feel like if it's confusing in the future we'll update and if it's not then it was the right call all along.

Reply:
ahh, yes, that makes sense but we cannot modify it. The plugin sdk expects that struct: https://github.com/grafana/grafana-plugin-sdk-go/blob/03484c4cb5e7b8ce6b7f8dfe8634540d74337517/data/time_series.go#L48